src/runtime_src/core/tools/xbtracer/src/lib/capture.cpp

```diff
@@ -0,0 +1,617 @@
+          symbol.st_name);
+      elf_file.read(symbol_name.data(), symbol_len);
+      if (!elf_file)
+      {
+        std::cerr << "Failed to read symbol name" << std::endl;
+        return 0;
+      }
+      // std::cout <<"Mangled name: "<<symbol_name.data() <<std::endl;
+      std::string demangled_name = demangle(symbol_name.data());
+      // std::cout <<"De-Mangled name: " << demangled_name << "\n";
+      func_mangled[demangled_name] = symbol_name.data();
+    }
+  }
+
+  return 1;
+}
+}  // namespace xrt::tools::xbtracer
+
+#elif _WIN32
+#include <Dbghelp.h>
+#include <windows.h>
+#pragma comment(lib, "Dbghelp.lib")
+
+namespace xrt::tools::xbtracer {
+  std::string demangle(const char* mangled)
+  {
+    constexpr const DWORD length = 512;  // Adjust the buffer size as needed
+    char demangled_str[length];
+
+    DWORD result = UnDecorateSymbolName(
+        mangled, demangled_str, length,
+        UNDNAME_NO_FUNCTION_RETURNS | UNDNAME_NO_ACCESS_SPECIFIERS |
+            UNDNAME_NO_ALLOCATION_LANGUAGE | UNDNAME_NO_ALLOCATION_MODEL |
+            UNDNAME_NO_MS_KEYWORDS | UNDNAME_NO_THROW_SIGNATURES);
+
+    if (result != 0)
+    {
+      std::vector<std::pair<std::string, std::string>> replacements =
+      {
+        { "class std::basic_string<char,struct std::char_traits<char>,"
+            "class std::allocator<char> >", "std::string" },
+        {"const ", "const"},
+        {"class ", ""},
+        {",", ", "},
+        {")const", ") const"},
+        {"__int64", "long"},
+        {"(void)", "()"},
+        {"enum ", ""},
+        {"struct std::ratio<1, 1000>", "std::ratio<1l, 1000l>"},
+        {"std::map<std::string, unsigned int, struct std::less<std::string >, "
+         "std::allocator<struct std::pair<std::string const, unsigned int> > >",
+             "xrt::hw_context::cfg_param_type"},
+        {"void *", "void*"}
+      };
+
+      std::string demangled_and_conditioned_str =
+          find_and_replace_all(demangled_str, replacements);
+      return demangled_and_conditioned_str;
+    } else {
+      return mangled;
+    }
+  }
+
+  // Make the page writable and replace the function pointer. Once
+  // replacement is completed restore the page protection.
+  static void replace_func(PIMAGE_THUNK_DATA thunk, void* func_ptr)
+  {
+    // Make page writable temporarily:
+    MEMORY_BASIC_INFORMATION mbinfo;
+    VirtualQuery(thunk, &mbinfo, sizeof(mbinfo));
+    if (!VirtualProtect(mbinfo.BaseAddress, mbinfo.RegionSize,
+                        PAGE_EXECUTE_READWRITE, &mbinfo.Protect))
+      return;
+
+    // Replace function pointer with our implementation:
+    thunk->u1.Function = (ULONG64)func_ptr;
+
+    // Restore page protection:
+    DWORD zero = 0;
+    if (!VirtualProtect(mbinfo.BaseAddress, mbinfo.RegionSize,
+                        mbinfo.Protect, &zero))
+      return;
+  }
+}  // namespace xrt::tools::xbtracer
+
+using namespace xrt::tools::xbtracer;
```

Comment:
Using does often make code more difficult to read.  You may want a namespace alias instead?
```suggestion
 namespace xtx = xrt::tools::xbtracer;
```

Reply:
Done as suggested, please check.